hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/server/UIClientCommWebSocket.java

```diff
@@ -0,0 +1,175 @@
+     * for.
+     *
+     * @param binaryDataStream contains the JSON request and additional binary data
+     * @param session the client session making the request
+     * @return the results of the command invocation; this is sent back to the UI client
+     */
+    @OnMessage
+    public String uiClientBinaryData(InputStream binaryDataStream, Session session) {
+        String uiClientId = getUIClientIDFromSession(session);
+        MsgLogger.LOG.infoReceivedBinaryDataFromUIClient(session.getId());
+
+        String requestClassName = "?";
+        BasicMessageWithExtraData<? extends BasicMessage> response = null;
+
+        try {
+            // parse the JSON and get its message POJO, including any additional binary data being streamed
+            BasicMessageWithExtraData<BasicMessage> reqWithData = new ApiDeserializer().deserialize(binaryDataStream);
+            BasicMessage request = reqWithData.getBasicMessage();
+            requestClassName = request.getClass().getName();
+
+            handleRequest(session, reqWithData);
+
+        } catch (WebsocketAuthenticationException wae) {
+            response = null;
+            try {
+                session.close(new CloseReason(CloseCodes.VIOLATED_POLICY, wae.getLocalizedMessage()));
+            } catch (IOException ioe) {
+                MsgLogger.LOG.errorf(ioe,
+                        "Failed to close UI client [%s] (session [%s]) after authentication failure (request=[%s])",
+                        uiClientId, session.getId(), requestClassName);
+            }
+        } catch (Throwable t) {
+            MsgLogger.LOG.errorCommandExecutionFailureUIClient(requestClassName, uiClientId, session.getId(), t);
+            String errorMessage = "Command failed [" + requestClassName + "]";
+            response = new BasicMessageWithExtraData<>(
+                    new GenericErrorResponseBuilder().setThrowable(t).setErrorMessage(errorMessage).build(), null);
+        }
+        if (response != null) {
+            try {
+                new WebSocketHelper().sendSync(session, response);
+            } catch (IOException ioe) {
+                MsgLogger.LOG.errorf(ioe, "Failed to send a message to UI client [%s] (session [%s]) (message=[%s])",
+                        uiClientId, session.getId(), response.getBasicMessage().getClass().getName());
+            }
+        }
+        return null;
+    }
+
+    @OnClose
+    public void uiClientSessionClose(Session session, CloseReason reason) {
+        String uiClientId = getUIClientIDFromSession(session);
+        MsgLogger.LOG.infoUISessionClosed(uiClientId, session.getId(), reason);
+        connectedUIClients.removeSession(session.getId(), session);
+    }
+
+    private String getUIClientIDFromSession(Session session) {
+        return session.getId(); // for now, the UI client ID *is* the session ID
+        // return session.getRequestParameterMap().get("Hawkular-UI-Client-ID").get(0);
```

Comment:
we can get rid of this commented out stuff and that "for now" comment. looks like session ID is going to be our client ID for the UI after all.